composer/trainer/trainer.py

```diff
@@ -887,6 +895,28 @@ def __init__(
             precision = Precision(precision)
         _validate_precision(precision, device)
 
+        # check if provided model is compiled or not
+        is_torch_2_0 = using_torch_2_0()
+        is_model_compiled = False
+        if is_torch_2_0:
+            from torch._dynamo import OptimizedModule
+            if isinstance(model, OptimizedModule):
+                log.warning(f'Provided `model` is already compiled with `torch.compile`. Ignoring ' +
+                            f'parameter `compile_config` if provided. If you would like `Trainer` ' +
+                            f'to takes care of model compilation, provide a un-compiled model and ' +
```

Comment:
```suggestion
                            f'to takes care of model compilation, provide a not-compiled model and ' +
```

Reply:
Done in https://github.com/mosaicml/composer/pull/2118/commits/d0e8481b1a3de2bb20239802a9adfac217fb45cc.